staging/src/k8s.io/client-go/util/workqueue/delaying_queue.go

```diff
@@ -31,6 +34,52 @@ type DelayingInterface interface {
 	Interface
 	// AddAfter adds an item to the workqueue after the indicated duration has passed
 	AddAfter(item interface{}, duration time.Duration)
+	// AddWithOptions allows full control over the delaying queue features.
+	AddWithOptions(item interface{}, opts DelayingOptions)
+	// ForgetWaiting will remove an item from the delayed/waiting queue - it won't cancel active queued items
+	// for them you are still expected to call Done()
+	ForgetWaiting(item interface{})
+}
+
+// DelayingOptions toggle implementation specific options for use with calls to AddWithOptions().
+type DelayingOptions struct {
+	// Duration specifies for how long this item should be delayed before being added back into the active queue.
+	Duration time.Duration
+	// Waiting specifies how you want the waitingLoop to insert items that are already waiting.
+	WhenWaiting AlreadyWaitingBehaviour
+	// DisableActiveWaitingDedup when true will enable these two use-case scenarios:
+	// 1. A item with a delay can be added regardless if the same item already exists in the active queue.
+	// 2. A waiting item will remain queued when the same item is immediately queued.
+	// The de-duplication between the two queues is performed by default.
+	DisableActiveWaitingDedup bool
```

Comment:
How about `PermitRedundancy` for a name?

Reply:
I think Permit is a positive change but I think that 'Redundancy' might be a little confusing for clients who come with pre-defined ideas about what Redundancy means.  How about PermitActiveWaiting?  Which feels like it evokes the fact the the item is/can be in two states.